Sources/Core/ObjectiveCDictionaryExtension.swift

```diff
@@ -0,0 +1,152 @@
+            ObjCIR.ifStmt("_"+"\(self.dirtyPropertiesIVarName).\(dirtyPropertyOption(propertyName: param, className: self.className))") {
+                [renderAddObjectStatement(param, schema, dictionary)]
+            }
+            }.joined(separator: "\n")
+        return ObjCIR.method("- (NSDictionary *)dictionaryRepresentation") {[
+            "NSMutableDictionary *\(dictionary) = [[NSMutableDictionary alloc]initWithCapacity:\(self.properties.count)];",
+            props,
+            "return dict;"
+            ]}
+    }
+}
+
+extension ObjCFileRenderer {
+    fileprivate func renderAddObjectStatement(_ param: String, _ schema: Schema, _ dictionary: String, counter: Int = 0) -> String {
+        let propIVarName = "_\(param.snakeCaseToPropertyName())"
+        switch schema {
+        case .boolean:
+            return "[\(dictionary) setObject: [NSNumber numberWithBool:"+propIVarName + "] forKey: @\"" + param + "\" ];"
+
+        case .float:
+            return "[\(dictionary) setObject: [NSNumber numberWithDouble:"+propIVarName + "] forKey: @\"" + param + "\" ];"
+        case .integer:
+            return "[\(dictionary) setObject: [NSNumber numberWithInteger:"+propIVarName + "] forKey: @\"" + param + "\" ];"
+        case .object:
+            return ObjCIR.scope {[
+                ObjCIR.ifStmt("\(propIVarName) != nil") {[
+                    "[\(dictionary) setObject: [\(propIVarName) dictionaryRepresentation]  forKey: @\"\(param)\" ];"
+                ]},
+                ObjCIR.elseStmt({[
+                    "[\(dictionary) setObject: [NSNull null] forKey: @\"" + param + "\" ];"
+                ]
+                })
+                ]}
+        case .string(format: .none),
+             .string(format: .some(.email)),
+             .string(format: .some(.hostname)),
+             .string(format: .some(.ipv4)),
+             .string(format: .some(.ipv6)):
+            return ObjCIR.scope {[
+                ObjCIR.ifStmt("\(propIVarName) != nil") {[
+                    "[\(dictionary) setObject: "+propIVarName + " forKey: @\"" + param + "\" ];"
+                    ]},
+                ObjCIR.elseStmt({[
+                    "[\(dictionary) setObject: [NSNull null] forKey: @\"" + param + "\" ];"
+                    ]
+                })
+                ]}
+        case .string(format: .some(.uri)):
+            return ObjCIR.scope {[
+                ObjCIR.ifStmt("\(propIVarName) != nil") {[
+                    "[\(dictionary) setObject: ["+propIVarName + " absoluteString] forKey: @\"" + param + "\" ];"
+                    ]},
+                ObjCIR.elseStmt({[
+                    "[\(dictionary) setObject: [NSNull null] forKey: @\"" + param + "\" ];"
+                    ]
+                })
+                ]}
+        case .string(format: .some(.dateTime)):
+            return ObjCIR.scope {[
+                ObjCIR.ifStmt("\(propIVarName) != nil") {[
```

Comment:
If we can use the transformer we should call `reversedTransformedValue:`. https://developer.apple.com/reference/foundation/nsvaluetransformer/1402020-reversetransformedvalue?language=objc